magento/pwa-studio
# #1617 - Fetch store name from the GraphQL server at compile time

## Description

With the changes from this PR, the `STORE_NAME` is not hardcoded anymore from the webpack's DefinePlugin 

https://github.com/magento/pwa-studio/blob/70cf1e0d1d656487f8e03d6aaaec21e78f09fb36/packages/venia-concept/webpack.config.js#L66

As suggested in #1617, if `STORE_VIEW_CODE` environment variable is used, the store_name value from the corresponding available store is used. If not, use the default value returned by the `getStoreConfigData` query is used for the `STORE_NAME`.

## Related Issue
Closes #1617.

## Acceptance




### Verification Stakeholders

@revanth0212 

### Specification



Changes to the `venia-concept` and `pwa-buildback` packages must be reviewed.

### Verification Steps



Setup and run the webserver.

1. Go to the homepage.
2. View page's source code.
3. Find the `title` tag
4. Make sure the `title` tag value contains the store name as specified by the `STORE_VIEW_CODE` (or the default store).

## Screenshots / Screen Captures (if appropriate)

![image](https://user-images.githubusercontent.com/13456702/108502404-6c846980-72bb-11eb-92ba-03fb63056678.png)


## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
